backend-elasticsearch/src/main/java/org/hibernate/search/v6poc/backend/elasticsearch/logging/impl/Log.java

```diff
@@ -71,36 +70,38 @@ void executedRequest(String method, String path, Map<String, String> getParamete
 
 	@Message(id = 502, value = "A search query cannot target both an Elasticsearch index and other types of index."
 			+ " First target was: '%1$s', other target was: '%2$s'" )
-	SearchException cannotMixElasticsearchSearchTargetWithOtherType(IndexSearchTargetBuilder firstTarget, ElasticsearchIndexManager otherTarget);
+	SearchExceptionWithContext cannotMixElasticsearchSearchTargetWithOtherType(IndexSearchTargetBuilder firstTarget,
+			ElasticsearchIndexManager otherTarget, @Param FailureContext context);
 
 	@Message(id = 503, value = "A search query cannot target multiple Elasticsearch backends."
 			+ " First target was: '%1$s', other target was: '%2$s'" )
-	SearchException cannotMixElasticsearchSearchTargetWithOtherBackend(IndexSearchTargetBuilder firstTarget, ElasticsearchIndexManager otherTarget);
+	SearchExceptionWithContext cannotMixElasticsearchSearchTargetWithOtherBackend(IndexSearchTargetBuilder firstTarget,
+			ElasticsearchIndexManager otherTarget, @Param FailureContext context);
 
-	@Message(id = 504, value = "Unknown field '%1$s' in indexes %2$s." )
-	SearchException unknownFieldForSearch(String absoluteFieldPath, Collection<String> indexNames);
+	@Message(id = 504, value = "Unknown field '%1$s'." )
+	SearchExceptionWithContext unknownFieldForSearch(String absoluteFieldPath, @Param FailureContext context);
 
-	@Message(id = 505, value = "Multiple conflicting types for field '%1$s': '%2$s' in index '%3$s', but '%4$s' in index '%5$s'." )
-	SearchException conflictingFieldTypesForSearch(String absoluteFieldPath,
-			ElasticsearchIndexSchemaFieldNode schemaNode1, String indexName1,
-			ElasticsearchIndexSchemaFieldNode schemaNode2, String indexName2);
+	@Message(id = 505, value = "Multiple conflicting types for field '%1$s': '%2$s' vs. '%3$s'." )
+	SearchExceptionWithContext conflictingFieldTypesForSearch(String absoluteFieldPath,
+			ElasticsearchIndexSchemaFieldNode schemaNode1, ElasticsearchIndexSchemaFieldNode schemaNode2,
+			@Param FailureContext context);
```

Comment:
Not sure it's an improvement here. You lose the link between the node and the index. But maybe it's not a big deal if it brings consistency.

Reply:
Yeah, I don't think it's very important since, in order to fix this error, you will very probably have to have a look at both indexes anyway.
Besides, we still display the nodes in the same order as index names, so even if it's not very clear, users could guess which node is in which index, and they would guess right.